stellargraph/data/explorer.py

```diff
@@ -796,3 +800,202 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
+            step_type <string>: whether the next step taken in the walk is chosen uniformly at random or is biased (such as, exponential). 
+            The default is 'uniform_random', any other string value assumes a biased time respecting walk. 
+            And currently only exponentially biased random walks are implemented.  
+
+        Returns:
+            <list> List of lists of nodes ids for each of the random walks
+
+        """
+        self._check_common_parameters(nodes, n, length, seed)
+        # self._check_temporal_parameters(bidirectional, edge_time_label)
+        rs = self._get_random_state(seed)
+
+        walks = []
+        for node in nodes:  # iterate over root nodes
+            for walk_number in range(n):  # generate n walks per root node
+                walk = self._walk(
+                    node, length, bidirectional, edge_time_label, step_type, rs
+                )
+                walks.append(walk)
+        return walks
+
+    def _check_temporal_parameters(self, bidirectional, edge_time_label):
+        """
+            Checks that the parameter values are valid or raises ValueError exceptions with a message indicating the
+            parameter (the first one encountered in the checks) with invalid value.
+    
+            Args:
+                bidirectional: <False or True> Indicates whether the temporal walk is in forward direction only or extends in both directions.
+                edge_time_label: <string> Label of the edge weight property.
+           """
+
+        if type(bidirectional) != bool:
+            self._raise_error(
+                "Parameter bidirectional has to be either False (temporal random walks forward in times) or True (temporal random walks extending in both directions)."
+            )
+
+        if not isinstance(edge_time_label, str):
+            self._raise_error("The edge time property label has to be of type string")
+
+        "Check that all edges have the time label specified by 'edge_time_label' and all the time values positive real values."
+        for node in self.graph.nodes():
+
+            for neighbor in self.graph.neighbors(
+                node, include_edge_weight=True, edge_types=[edge_time_label]
+            ):
+                for n, t in neighbor:
+                    if not isinstance(t, (int, float)):
+                        self._raise_error(
+                            "Timestamp between nodes ({}) and ({}) is not numeric ({}).".format(
+                                node, neighbor, t
+                            )
+                        )
+                    elif t < 0:  # check if edge has a negative timestamp
+                        self._raise_error(
+                            "Timestamp between nodes ({}) and ({}) is negative ({}).".format(
+                                node, neighbor, t
+                            )
+                        )
+
+    def get_exp_distribution(self, times, current_time, orientation="forward"):
```

Comment:
`orientation` is unused. I might rename this to `is_forward` for consistency, and try figure out how I should use it (I'm guessing all the `t_m - t_n` get reversed)

Reply:
actually there's no point in reversing since all time distances are wrapped in`np.abs`, so I'm guessing it's the same regardless of direction

Reply:
actually^2 maybe I'm supposed to use `max(times)` instead of `min(times)` for going backwards... will just double check the paper